tests/test_rainflow.py

```diff
@@ -35,8 +53,56 @@ def test_rainflow_ndigits(self):
         self.assertNotEqual(rainflow.count_cycles(series), self.cycles)
         self.assertEqual(rainflow.count_cycles(series, ndigits=1), self.cycles)
 
+    def test_rainflow_nbins(self):
+        self.assertEqual(
+            rainflow.count_cycles(self.series, left=True, right=True, nbins=1),
+            self.cycles_nbins_1,
+        )
+        self.assertEqual(
+            rainflow.count_cycles(self.series, left=True, right=True, nbins=2),
+            self.cycles_nbins_2,
+        )
+        self.assertEqual(
+            rainflow.count_cycles(self.series, left=True, right=True, nbins=5),
+            self.cycles_nbins_5,
+        )
+        self.assertEqual(
+            rainflow.count_cycles(self.series, left=True, right=True, nbins=9),
+            self.cycles_nbins_9,
+        )
+        self.assertEqual(
+            rainflow.count_cycles(self.series, left=True, right=True, nbins=10),
+            self.cycles_nbins_10,
+        )
+
+    def test_rainflow_binsize(self):
+        self.assertEqual(
+            rainflow.count_cycles(self.series, left=True, right=True, binsize=1),
+            self.cycles_binsize_1,
+        )
+        self.assertEqual(
+            rainflow.count_cycles(self.series, left=True, right=True, binsize=2),
+            self.cycles_binsize_2,
+        )
+        self.assertEqual(
+            rainflow.count_cycles(self.series, left=True, right=True, binsize=3),
+            self.cycles_binsize_3,
+        )
+        self.assertEqual(
+            rainflow.count_cycles(self.series, left=True, right=True, binsize=5),
+            self.cycles_binsize_5,
+        )
+        self.assertEqual(
+            rainflow.count_cycles(self.series, left=True, right=True, binsize=9),
+            self.cycles_binsize_9,
+        )
+        self.assertEqual(
+            rainflow.count_cycles(self.series, left=True, right=True, binsize=10),
+            self.cycles_binsize_10,
+        )
+
     def test_series_with_zero_derivatives(self):
-    	series = itertools.chain(*([x, x] for x in self.series))
+    	series = list(itertools.chain(*([x, x] for x in self.series)))
```

Comment:
Let's revert to the original test. `count_cycles` was working with iterators so let's not change that.

Reply:
ok